src/halmos/__main__.py

```diff
@@ -21,7 +21,12 @@
 if hasattr(sys, 'set_int_max_str_digits'): # Python verion >=3.8.14, >=3.9.14, >=3.10.7, or >=3.11
     sys.set_int_max_str_digits(0)
 
-def parse_args(args) -> argparse.Namespace:
+def mk_crytic_parser() -> argparse.ArgumentParser:
+    crytic_compile_parser = argparse.ArgumentParser(prog='crytic-compile')
```

Comment:
Which program name do we want to show in the --compile-help message: `crytic-compile` vs `halmos`?  The current PR version prints as follows:
```
$ halmos tests --compile-help
usage: crytic-compile [-h] [--compile-force-framework COMPILE_FORCE_FRAMEWORK] ...
       ^^^^^^^^^^^^^^
```

Reply:
You're right, that usage string is out of place. I removed it from the output in 5383aa6:

```
halmos --help-compile 2>&1 | head                                                                                                                                                                                                              

Compile options:
  --compile-force-framework COMPILE_FORCE_FRAMEWORK
                        Force the compile to a given framework (solc,truffle,e
                        mbark,dapp,etherlime,etherscan,vyper,waffle,brownie,so
                        lc-json,buidler,hardhat,foundry,standard,archive)
  --compile-remove-metadata
                        Remove the metadata from the bytecodes
  --compile-custom-build COMPILE_CUSTOM_BUILD
                        Replace platform specific build command
  --ignore-compile      Do not run compile of any platform
```